11-Backbonejs/06-Adding-Custom-Events.md

```diff
@@ -0,0 +1,200 @@
+
+The `trigger` function causes all callbacks to the given Event to run.  You can specify an argument to pass to the callback function as a 2nd argument.  In the example above we did so here:  `task.trigger("Past_Due", {msg: "past due",
+                            date: new Date()                        
+});`.
+
+
+
+## Adding Events To A View
+
+There are two types of Events a View can listen for in Backbone.  The first are DOM events like we've dealt with earlier.  The second are custom events using Backbone's Event API like we've seen above.  
+
+DOM events are bound using the View's `events` property or using `jQuery.on()`.  Look at [this CodePen](http://codepen.io/CheezItMan/pen/GNvMVG?editors=1010).  In this CodePen we have created a view for a Task item and added an event when the view is clicked on.
+
+Dom Events are added as an Object property of the view with the name of the event as the "key" and the value being the name of the function which serves as the callback function for the event.  
+
+```javascript
+var TodoView = Backbone.View.extend({
+  tagName: "li",
+  events: {
+    "click": "onClick"
+  },
+  onClick: function() {
+    console.log("Clicked!  Model:  " + this.model.get("title"));
+  },
+  render: function() {
+    this.$el.html(this.model.get("title"));
+    return this;
+  }
+});
+``` 
+
+If you wanted to listen for a specific item within the view you could, just like with jQuery specify the item with a CSS Selector.  For example if there was a block within the view with a class `title` and you wanted to listen for clicks on that block, you could put this line into the events hash `"click .title": "onTitleClick"`.  This would cause the View to listen for clicks on any element within the view and with the `title` class.  When any sub-element with that class is clicked on the `onTitleClick` event handler will run.
+
+
+Backbone API Events can be added with the `.on` method as you saw earlier.  This lets you create custom events that can be triggered in your Models or Views to update other objects when something occurs, a change in status, user interaction etc.  
+
+Looking at the TodoViews in the Codepen, we can see when a task is clicked on using a DOM event.  However it would be REALLY useful if we could update another view when a Todo item is clicked on, for example to show more details on the selected Todo item.  Thus we return to the Backbone Events API and the idea of an Event Bus.
+
+### listenTo 
+
+Similar to the `.on` function, the `.listenTo` method lets a Backbone object listen to an event that occurs on another object.  Look below and see how `listenTo` differs from the `on` function.  The main difference between the two is the context in which they add callbacks.  `on` lets you directly add a callback function to another object.  `listenTo` tells an object to listen for events on another object.  This allows the 'listener' to keep track of the objects for which it is listening.  
```

Comment:
I'll add more, plus when a view is destroyed `listenTo()` will automatically remove all event listeners, while on will not.  So you **can** end up with zombie listeners out there.